qemu/tests/rebase_onto_no_backing_file.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import json
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+from qemu.tests.qemu_disk_img import QemuImgTest
+from qemu.tests.qemu_disk_img import generate_base_snapshot_pair
+
+
+def run(test, params, env):
+    """
+    Rebase a qcow2 snapshot onto no backing file.
+
+    1. create a qcow2 snapshot based on a raw image with compat mode 0.10/1.1
+    2. boot the guest from the base
+    3. create a file in the base disk, calculate its md5sum
+    4. shut the guest down
+    5. rebase the snapshot to null to remove the raw backing file
+    6. check the snapshot
+    7. boot the guest from the snapshot and check whether the
+        file's md5sum stays same
+
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def _get_img_obj_and_params(tag):
+        """Get an QemuImg object and its params based on the tag."""
+        img_param = params.object_params(tag)
+        img = QemuImg(img_param, data_dir.get_data_dir(), tag)
+        return img, img_param
+
+    def _get_compat_version():
+        """Get snapshot compat version."""
+        if params.get("image_extra_params") is None:
+            # default compat version for now is 1.1
+            return "1.1"
+        return params.get("image_extra_params").split("=")[1]
+
+    def _verify_qemu_img_info(output, b_fmt, b_name):
+        """Verify qemu-img info output for this case."""
+        logging.info("Verify snapshot's backing file information.")
+        res = json.loads(output)
+        if (res["backing-filename-format"] != b_fmt or
+                res["backing-filename"] != b_name):
+            test.fail("Backing file information is not correct,"
+                      " got %s." % b_name)
+        compat = res["format-specific"]["data"]["compat"]
+        expected = _get_compat_version()
+        if (compat != expected):
+            test.fail("Snapshot's compat mode is not correct,"
+                      " got %s, expected %s." % (compat, expected))
+
+    def _verify_no_backing_file(output):
+        """Verify snapshot has no backing file for this case."""
+        logging.info("Verify snapshot has no backing file after rebase.")
+        for k in json.loads(output):
+            if "backing" in k:
+                test.fail("The snapshot has backing file after rebase.")
+
+    file = params["guest_file_name"]
+    gen = generate_base_snapshot_pair(params["image_chain"])
+    base, snapshot = next(gen)
+    base_img, _ = _get_img_obj_and_params(base)
+    sn_img, sn_img_params = _get_img_obj_and_params(snapshot)
+
+    logging.info("Create a snapshot %s based on %s." % (snapshot, base))
```

Comment:
could just use `logging.info("Create a snapshot %s based on %s.",  snapshot, base)`, `info` will take care of the %-formatting.